platform/darwin/src/NSPredicate+MGLAdditions.mm

```diff
@@ -0,0 +1,63 @@
+#import "NSPredicate+MGLAdditions.h"
+
+class FilterEvaluator {
+public:
+    
+    mbgl::Value getValue(id obj) {
+        if ([obj isKindOfClass:NSString.class]) {
+            NSString *string = (NSString *)obj;
+            return std::string(string.UTF8String);
+        } else if ([obj isKindOfClass:NSString.class]) {
+            NSNumber *number = (NSNumber *)obj;
+            if((strcmp([number objCType], @encode(int))) == 0) {
+                return number.intValue;
+            } else if ((strcmp([number objCType], @encode(float))) == 0) {
+                return number.floatValue;
+            } else {
+                return number.boolValue;
+            }
+        }
+        return "";
+    }
+    
+    mbgl::style::Filter operator()(NSComparisonPredicate *predicate) {
+        const NSPredicateOperatorType operatorType = predicate.predicateOperatorType;
+        switch (operatorType) {
+            case NSEqualToPredicateOperatorType: {
+                auto filter = mbgl::style::EqualsFilter();
+                filter.key = predicate.leftExpression.keyPath.UTF8String;
+                filter.value = getValue(predicate.rightExpression.constantValue);
+                return filter;
+            }
+            case NSNotEqualToPredicateOperatorType: {
+                auto filter = mbgl::style::NotEqualsFilter();
+                filter.key = predicate.leftExpression.keyPath.UTF8String;
+                filter.value = getValue(predicate.rightExpression.constantValue);
+                return filter;
+            }
+            default:
```

Comment:
Instead of a `default` case that raises, add cases for all of the `NSPredicateOperatorType` values. If Foundation ever adds a new `NSPredicateOperatorType` value, we’ll get a compiler warning, but not if there’s a `default` case.